packages/backend/src/services/ProjectService/ProjectService.ts

```diff
@@ -2656,16 +2676,28 @@ export class ProjectService extends BaseService {
             user_uuid: user.userUuid,
         };
         const schema = ProjectService.getWarehouseSchema(credentials);
+        const database = ProjectService.getWarehouseDatabase(credentials);
+
+        if (!schema) {
+            throw new NotFoundError(
+                'Schema not found in warehouse credentials',
+            );
+        }
+        if (!database) {
+            throw new NotFoundError(
+                'Database not found in warehouse credentials',
+            );
+        }
 
-        const warehouseCatalog = warehouseClient.getFields(
+        const warehouseCatalog = await warehouseClient.getFields(
             tableName,
             schema,
             queryTags,
         );
 
         await sshTunnel.disconnect();
 
-        return warehouseCatalog;
+        return warehouseCatalog[database][schema][tableName];
```

Comment:
now we get the `database` depending on the warehouse. 
LMK what you think @rephus and if you would do it differently.